dev/PushNotifications/PushNotificationReceivedEventArgs.cpp

```diff
@@ -23,49 +23,105 @@ namespace winrt
+    {
+        return { payload, payload + (payloadLength * sizeof(uint8_t)) };
+    }
 
     winrt::com_array<uint8_t> PushNotificationReceivedEventArgs::Payload()
     {
-        auto rawNotificationData = m_rawNotification.data();
-        return { rawNotificationData, rawNotificationData + (m_rawNotification.Length() * sizeof(uint8_t)) };
+        return { m_rawNotificationPayload.data(), m_rawNotificationPayload.data() + (m_rawNotificationPayload.size() * sizeof(uint8_t)) };
     }
 
     winrt::BackgroundTaskDeferral PushNotificationReceivedEventArgs::GetDeferral()
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
-
-        return m_backgroundTaskInstance.GetDeferral();
+        if (m_isUnpackagedApp)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
+
+            return m_backgroundTaskInstance.GetDeferral();
+        }
+        else
+        {
+            return nullptr;
+        }
     }
 
     winrt::event_token PushNotificationReceivedEventArgs::Canceled(winrt::BackgroundTaskCanceledEventHandler const& handler)
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
-
-        return m_backgroundTaskInstance.Canceled(handler);
+        if (m_isUnpackagedApp)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
+
+            return m_backgroundTaskInstance.Canceled(handler);
+        }
+        else
+        {
+            return { 0 };
+        }
     }
 
     void PushNotificationReceivedEventArgs::Canceled(winrt::event_token const& token) noexcept
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
+        if (m_isUnpackagedApp)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_backgroundTaskInstance, "Foreground activation cannot call this.");
 
-        m_backgroundTaskInstance.Canceled(token);
+            m_backgroundTaskInstance.Canceled(token);
+        }
     }
 
     bool PushNotificationReceivedEventArgs::Handled()
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_args, "Background activation cannot call this.");
-
-        return m_args.Cancel();
+        if (m_isUnpackagedApp)
+        {
+            THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_args, "Background activation cannot call this.");
+
+            return m_args.Cancel();
+        }
+        else
+        {
+            return m_handledUnpackaged;
+        }
     }
 
     void PushNotificationReceivedEventArgs::Handled(bool value)
     {
-        THROW_HR_IF_NULL_MSG(E_ILLEGAL_METHOD_CALL, m_args, "Background activation cannot call this.");
-
-        m_args.Cancel(value);
+        if (m_isUnpackagedApp)
```

Comment:
I'm really confused with m_unpackagedApp Vs m_unpackagedAppScenario. I think you mean the latter since packaged apps can also go through the unpackaged scenario. I would reword all the variables to clearly indicate the context.